src/sqllogictest/src/runner.rs

```diff
@@ -202,25 +202,39 @@ impl fmt::Display for Outcome<'_> {
     }
 }
 
-#[derive(Default, Debug, Eq, PartialEq)]
-pub struct Outcomes([usize; NUM_OUTCOMES]);
+#[derive(Debug, Eq, PartialEq)]
+pub struct Outcomes {
+    inner: [usize; NUM_OUTCOMES],
+    skip_failures: bool,
```

Comment:
So I don't think`skip_failures` belongs here. It's a property of how `Outcomes` is *displayed*, not an inherent property of the outcome. This tension is on display in the implementation of `ops::AddAssign`: what does it mean to add together two `Outcomes` that have different `skip_failures` settings?

Reply:
> what does it mean to add together two Outcomes that have different skip_failures settings?

In the fullness of time it would mean that we can mix expected failures and not-expected-failures in the same process, but I agree right now it doesn't work.